stress/v2/statement/query.go

```diff
@@ -0,0 +1,157 @@
+package statement
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/influxdata/influxdb/models"
+	"github.com/influxdata/influxdb/stress/v2/ponyExpress"
+)
+
+// QueryStatement is a Statement Implementation to run queries on the target InfluxDB instance
+type QueryStatement struct {
+	StatementID string
+	Name        string
+
+	// TemplateString is a query template that can be filled in by Args
+	TemplateString string
+	Args           []string
+
+	// Number of queries to run
+	Count int
+
+	// Tracer for tracking returns
+	Tracer *ponyExpress.Tracer
+
+	// track time for all queries
+	runtime time.Duration
+}
+
+// This function adds tags to the recording points
+func (i *QueryStatement) tags() map[string]string {
+	tags := make(map[string]string)
+	return tags
+}
+
+// SetID statisfies the Statement Interface
+func (i *QueryStatement) SetID(s string) {
+	i.StatementID = s
+}
+
+// Run statisfies the Statement Interface
+func (i *QueryStatement) Run(s *ponyExpress.StoreFront) {
+
+	// Set the tracer
+	i.Tracer = ponyExpress.NewTracer(i.tags())
+
+	vals := make(map[string]interface{})
+
+	var point models.Point
+
+	runtime := time.Now()
+
+	for j := 0; j < i.Count; j++ {
+
+		// If the query is a simple query, send it.
+		if len(i.Args) == 0 {
+			b := []byte(i.TemplateString)
+
+			// Make the package
+			p := ponyExpress.NewPackage(ponyExpress.Query, b, i.StatementID, i.Tracer)
+
+			// Increment the tracer
+			i.Tracer.Add(1)
+
+			// Send the package
+			s.SendPackage(p)
+
+		} else {
+			// Otherwise cherry pick field values from the commune?
+
+			// TODO: Currently the program lock up here if s.GetPoint
+			//       cannot return a value, which can happen.
+			// Seee insert.go
+			s.Lock()
+			point = s.GetPoint(i.Name, s.Precision)
+			s.Unlock()
+
+			setMapValues(vals, point)
+
+			// Set the template string with args from the commune
+			b := []byte(fmt.Sprintf(i.TemplateString, setArgs(vals, i.Args)...))
+
+			// Make the package
+			p := ponyExpress.NewPackage(ponyExpress.Query, b, i.StatementID, i.Tracer)
+
+			// Increment the tracer
+			i.Tracer.Add(1)
+
+			// Send the package
+			s.SendPackage(p)
+
+		}
+	}
+
+	// Wait for all operations to finish
+	i.Tracer.Wait()
+
+	// Stop time timer
+	i.runtime = time.Since(runtime)
+}
+
+// Report statisfies the Statement Interface
+func (i *QueryStatement) Report(s *ponyExpress.StoreFront) string {
+	// Pull data via StoreFront client
+	allData := s.GetStatementResults(i.StatementID, "query")
+
+	if allData == nil || allData[0].Series == nil {
```

Comment:
If `allData` is not `nil`, but is empty, this will panic. Perhaps `if len(allData) == 0 || allData[0].Series == nil` is preferable.